File: languages/python/bloock/entity/key/ecdsa_key_pair.py
from bloock._bridge.proto.keys_pb2 import GenerateLocalKeyResponse
from bloock.entity.key.key_pair import KeyPair


class EcdsaKeys(KeyPair):
    @staticmethod
    def from_proto(res: GenerateLocalKeyResponse) -> KeyPair:
        return EcdsaKeys(res.local_key.key, res.local_key.private_key)

File: languages/php/Bloock/Entity/Key/EcdsaKeyPair.php
<?php

namespace Bloock\Entity\Key;

use Bloock\GenerateLocalKeyResponse;

/**
 * Represents an ecdsa key pair, with private and public key.
 */
class EcdsaKeyPair extends KeyPair
{
    public function __construct(string $publicKey, string $privateKey)
    {
        parent::__construct($publicKey, $privateKey);
    }

    public static function fromProto(GenerateLocalKeyResponse $res): EcdsaKeyPair
    {
        return new EcdsaKeyPair($res->getLocalKey()->getKey(), $res->getLocalKey()->getPrivateKey());
    }
}